adam-core/src/main/scala/org/bdgenomics/adam/models/SingleReadBucket.scala

```diff
@@ -49,6 +53,114 @@ case class SingleReadBucket(primaryMapped: Iterable[AlignmentRecord] = Seq.empty
   def allReads = {
     primaryMapped ++ secondaryMapped ++ unmapped
   }
+
+  def toFragment: Fragment = {
+    // take union of all reads, as we will need this for building and
+    // want to pay the cost exactly once
+    val unionReads = allReads
+
+    // do we have any chimeric reads?
+    val sequences: Seq[Sequence] = if (secondaryMapped.exists(_.getSupplementaryAlignment)) {
```

Comment:
Also, should we refactor `SingleReadBucket` to split secondary and supplementary reads apart?